File: src/test/java/com/ject/studytrip/member/helper/MemberTestHelper.java
package com.ject.studytrip.member.helper;

import com.ject.studytrip.member.domain.model.Member;
import com.ject.studytrip.member.domain.repository.MemberRepository;
import com.ject.studytrip.member.fixture.MemberFixture;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class MemberTestHelper {
    private final MemberRepository memberRepository;

    @Autowired
    public MemberTestHelper(MemberRepository memberRepository) {
        this.memberRepository = memberRepository;
    }

    public Member saveMember() {
        Member member = MemberFixture.createMemberFromKakao();
        return memberRepository.save(member);
    }

    public Member saveMember(String email, String nickname) {
        Member member = MemberFixture.createMemberFromKakao(email, nickname);
        return memberRepository.save(member);
    }

    public void deleteMemberById(Long memberId) {
        memberRepository.deleteById(memberId);
    }
}

File: src/test/kotlin/com/ject/studytrip/member/helper/MemberTestHelper.kt
package com.ject.studytrip.member.helper

import com.ject.studytrip.member.domain.model.Member
import com.ject.studytrip.member.domain.repository.MemberRepository
import com.ject.studytrip.member.fixture.MemberFixture
import org.springframework.stereotype.Component

@Component
class MemberTestHelper(
    private val memberRepository: MemberRepository,
) {
    fun saveMember(): Member = memberRepository.save(MemberFixture().createFromKakao())

    fun saveNewMember(
        email: String,
        nickname: String,
    ): Member = memberRepository.save(MemberFixture().createFromKakao(email, nickname))

    fun saveDeletedMember(
        email: String,
        nickname: String,
    ): Member = memberRepository.save(MemberFixture().createFromKakao(email, nickname).also { it.updateDeletedAt() })

    fun deleteMemberById(id: Long) = memberRepository.deleteById(id)
}
